File: documentation-examples-java/src/java/dev/typr/foundations/docs/sqlite/NullableTypes.java
package dev.typr.foundations.docs.sqlite;

import dev.typr.foundations.SqliteType;
import dev.typr.foundations.SqliteTypes;
import java.util.Optional;

@SuppressWarnings("unused")
public class NullableTypes {
  // start
  SqliteType<Long> notNull = SqliteTypes.integer;
  SqliteType<Optional<Long>> nullable = SqliteTypes.integer.opt();
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/sqlite/NullableTypes.scala
package dev.typr.foundationssc.docs.sqlite
import dev.typr.foundationssc.*

@SuppressWarnings(Array("unused"))
object NullableTypes:
  // start
  val notNull: SqliteType[Long] = SqliteTypes.integer
  val nullable: SqliteType[Option[Long]] = SqliteTypes.integer.opt
  // stop
